components/deprecated.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { mapValues } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	ifCondition,
+	withGlobalEvents,
+	withInstanceId,
+	withSafeTimeout,
+	withState,
+} from '@wordpress/compose';
+import deprecated from '@wordpress/deprecated';
+
+const deprecatedFunctions = {
+	ifCondition,
+	withGlobalEvents,
+	withInstanceId,
+	withSafeTimeout,
+	withState,
+};
+
+export default mapValues( deprecatedFunctions, ( deprecatedFunction, key ) => {
+	return ( ...args ) => {
+		deprecated( 'wp.components.' + key, {
```

Comment:
Historically our `deprecated` ESLint rule has stumbled spectacularly on any sort of cleverness. Did we check to see how this works?

Also, with "complications" like the need to remove exports from `index.js` which is not entirely obvious by this deprecation call alone, it would be good to have included a comment noting said needs to the future maintainer.